test/ectest.c

```diff
@@ -1679,9 +1679,8 @@ static int check_named_curve_test(int id)
                                              group_cofactor))
         || !TEST_int_eq(EC_GROUP_check_named_curve(gtest, 0, NULL), 0)
         /* The order is not an optional field, so this should fail */
-        || !TEST_true(EC_GROUP_set_generator(gtest, group_gen, NULL,
+        || TEST_true(EC_GROUP_set_generator(gtest, group_gen, NULL,
```

Comment:
nit: spacing of following line is off by 1

Reply:
This unit test is `master` only currently so it is fine to modify it and make sure a failure is returned if the mandatory `order` argument is not valid (`NULL`).

We lose a little bit of coverage on `EC_GROUP_check_named_curve()` but now, rationalizing the behaviour of `EC_GROUP_set_generator()`, the only way the former can process an `EC_GROUP` that has `order == 0` is if the `EC_GROUP` has been created from scratch and the generator is never set before calling `EC_GROUP_check_named_curve()`.

This case basically means that the `EC_GROUP` has not been properly initialized before calling `EC_GROUP_check_named_curve()` but I don't think we test any other function for this corner case condition.

@slontis do you think it's worth it to add a separate check to test `EC_GROUP_check_named_curve()`'s behaviour in this corner case?

Reply:
> nit: spacing of following line is off by 1

fixed in 23540664ac7e2361fe3bb191a28a21c0cd7890f8

Reply:
>do you think it's worth it to add a separate check to test EC_GROUP_check_named_curve()'s behaviour in this corner case?

Its probably not needed.